source/kraken/tests/disruption_reader_test.cpp

```diff
@@ -690,3 +732,262 @@ BOOST_AUTO_TEST_CASE(two_properties) {
+
+    const_it.set_message("m_id_3", "message_text_notification", "1", "2");
+    const_it.set_channel("channel_id_3", "channel_name_notification", "type","400" ,"1", "2");
+    const_it.set_channel_type("id_3", "notification");
+    reader(const_it);
+
+    // Duplicate combination of message, channel and channel_type
+    const_it.set_application_period("0", "1", "2");
+    const_it.set_ptobject("id_1", "uri_1", "line", "1", "2");
+    const_it.set_message("m_id_1", "message_text_web", "1", "2");
+    const_it.set_channel("channel_id", "channel_name_web", "type","400" ,"1", "2");
+    const_it.set_channel_type("id_1", "web");
+    reader(const_it);
+    const_it.set_application_period("1", "2", "3");
+    const_it.set_ptobject("id_2", "uri_2", "line", "1", "2");
+    const_it.set_message("m_id_2", "message_text_mobile", "1", "2");
+    const_it.set_channel("channel_id_2", "channel_name_mobile", "type","400" ,"1", "2");
+    const_it.set_channel_type("id_2", "mobile");
+    reader(const_it);
+    const_it.set_message("m_id_3", "message_text_notification", "1", "2");
+    const_it.set_channel("channel_id_3", "channel_name_notification", "type","400" ,"1", "2");
+    const_it.set_channel_type("id_3", "notification");
+    reader(const_it);
+
+    const auto& disruption = reader.disruption;
+    BOOST_CHECK_EQUAL(disruption->id(), "1");
+    auto cause = disruption->cause();
+    BOOST_CHECK_EQUAL(cause.id(), "1");
+    BOOST_CHECK_EQUAL(disruption->tags_size(), 1);
+    BOOST_REQUIRE_EQUAL(disruption->impacts_size(), 1);
+    auto impact = disruption->impacts(0);
+    BOOST_CHECK_EQUAL(impact.id(), "1");
+    auto severity = impact.severity();
+    BOOST_CHECK_EQUAL(severity.id(), "2");
+    BOOST_CHECK_EQUAL(impact.application_periods_size(), 2);
+    BOOST_CHECK_EQUAL(impact.informed_entities_size(), 2);
+    BOOST_REQUIRE_EQUAL(impact.messages_size(), 3);
+    // Message web
+    auto message = impact.messages(0);
+    BOOST_CHECK_EQUAL(message.text(), "message_text_web");
+    auto channel = message.channel();
+    BOOST_CHECK_EQUAL(channel.name(), "channel_name_web");
+    BOOST_REQUIRE_EQUAL(channel.types_size(), 1);
+    auto channel_type = channel.types(0);
+    BOOST_CHECK_EQUAL(channel_type, chaos::Channel_Type_web);
+    // Message mobile
+    message = impact.messages(1);
+    BOOST_CHECK_EQUAL(message.text(), "message_text_mobile");
+    channel = message.channel();
+    BOOST_CHECK_EQUAL(channel.name(), "channel_name_mobile");
+    BOOST_REQUIRE_EQUAL(channel.types_size(), 1);
+    channel_type = channel.types(0);
+    BOOST_CHECK_EQUAL(channel_type, chaos::Channel_Type_mobile);
+
+    // Message notification
+    message = impact.messages(2);
+    BOOST_CHECK_EQUAL(message.text(), "message_text_notification");
+    channel = message.channel();
+    BOOST_CHECK_EQUAL(channel.name(), "channel_name_notification");
+    // Here we have 4 channel_types instead of 1. This is an error
+    BOOST_REQUIRE_EQUAL(channel.types_size(), 4);
```

Comment:
We don't want to enforce that behaviour : if we get a code that corrects it, it's OK, but then this test would fail and we wouldn't know if it's a problem.